src/RouterInterface.php

```diff
@@ -0,0 +1,48 @@
+<?php
+/**
+ * Starlit App.
+ *
+ * @copyright Copyright (c) 2016 Starweb AB
+ * @license   BSD 3-Clause
+ */
+
+namespace Starlit\App;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Routing\Route;
+use Symfony\Component\Routing\RouteCollection;
+use Symfony\Component\Routing\Exception\ResourceNotFoundException;
+
+interface RouterInterface
+{
+    public function setOptions(array $options): void;
+
+    public function getDefaultModule(): string;
+
+    public function getDefaultController(): string;
+
+    public function getDefaultAction(): string;
+
+    public function addRoute(Route $route, string $name = null): void;
+
+    public function clearRoutes(): void;
+
+    public function getRoutes(): RouteCollection;
+
+    /**
+     * Resolve controller action and return callable properties.
+     *
+     * @throws ResourceNotFoundException
+     */
+    public function route(Request $request): AbstractController;
```

Comment:
This is a good start, but should be better implmented against a contract (interface)